vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -1,1017 +1,1023 @@
-          return selectedKeys[e]
-        });
-      }
-    }
-
-    /**
-     * Updates the cache for the given page for grid or tree-grid.
-     *
-     * @param page index of the page to update
-     * @param parentKey the key of the parent item for the page
-     * @returns an array of the updated items for the page, or undefined if no items were cached for the page
-     */
-    const updateGridCache = function(page, parentKey) {
-      let items;
-      if((parentKey || root) !== root) {
-        items = cache[parentKey][page];
-        let parentCache = grid.$connector.getCacheByKey(parentKey);
-        if(parentCache && parentCache.itemkeyCaches) {
-          let _cache = parentCache.itemkeyCaches[parentKey];
-          _updateGridCache(page, items,
-            treePageCallbacks[parentKey][page],
-            _cache);
-        }
-
-      } else {
-        items = cache[root][page];
-        _updateGridCache(page, items, rootPageCallbacks[page], grid._cache);
-      }
-      return items;
-    };
-
-    const _updateGridCache = function(page, items, callback, levelcache) {
-      // Force update unless there's a callback waiting
-      if (!callback) {
-        let rangeStart = page * grid.pageSize;
-        let rangeEnd = rangeStart + grid.pageSize;
-        if (!items) {
-          if (levelcache && levelcache.items) {
-            for (let idx = rangeStart; idx < rangeEnd; idx++) {
-              delete levelcache.items[idx];
+        let detailsOpenedItems = Array.from(grid.detailsOpenedItems);
+        let updatedSelectedItem = false;
+        for (let i = 0; i < items.length; ++i) {
+          const item = items[i];
+          if(!item) {
+            continue;
+          }
+          if (item.detailsOpened) {
+            if(grid._getItemIndexInArray(item, detailsOpenedItems) < 0) {
+              detailsOpenedItems.push(item);
             }
+          } else if(grid._getItemIndexInArray(item, detailsOpenedItems) >= 0) {
+            detailsOpenedItems.splice(grid._getItemIndexInArray(item, detailsOpenedItems), 1)
+          }
+          if (selectedKeys[item.key]) {
+            selectedKeys[item.key] = item;
+            item.selected = true;
+            updatedSelectedItem = true;
+          }
+        }
+        grid.detailsOpenedItems = detailsOpenedItems;
+        if (updatedSelectedItem) {
+          // IE 11 Object doesn't support method values
+          grid.selectedItems = Object.keys(selectedKeys).map(function(e) {
+            return selectedKeys[e]
+          });
+        }
+      })
+
+      /**
+       * Updates the cache for the given page for grid or tree-grid.
+       *
+       * @param page index of the page to update
+       * @param parentKey the key of the parent item for the page
+       * @returns an array of the updated items for the page, or undefined if no items were cached for the page
+       */
+      const updateGridCache = tryCatchWrapper(function(page, parentKey) {
```

Comment:
This seems to only be called from within other methods that are already using `tryCatchWrapper` so it could be removed here.

Also `_updateGridCache` below is only called from here so that also doesn't need `tryCatchWrapper`.

Reply:
Updated.